### PR TITLE
Upgrade from python2 to python3

### PR DESCRIPTION
**Pull request by RedBrogdon**
* Changes App Engine python version to 3.7.
* Adds flask (required for python3)
* Rewrites handling logic to be flask based.

Staged at https://20200205t114450-dot-dart-pad.appspot.com/

### CODE REVIEW

**Comment by RedBrogdon** [comment]
@jcollins-g Mind giving this a quick once over?

CC: @johnpryan @domesticmouse

File: web/main.py [hunk 1]
```diff
@@ -1,109 +1,68 @@
...
+# called `app` in this file.
+app = Flask(__name__)
+
+if __name__ == '__main__':
+    # This is used when running locally only.
+    app.run(host='127.0.0.1', port=8080, debug=True)
+
+# Files that the server is allowed to serve. Additional static files are
+# served via directives in app.yaml.
+VALID_FILES= [
```
Comment (jcollins-g) [inline web/main.py]: Maybe not for something this simple, but consider frozenset() if this gets much bigger.

File: web/app.yaml [hunk 2]
```diff
@@ -1,44 +1,39 @@
...
-- url: .*
-  script: main.app
-# Enable for version == prod.
-  secure: always
-
-libraries:
-- name: webapp2
-  version: "2.5.2"
-
-skip_files:
```
Comment (domesticmouse) [inline web/app.yaml]: Why drop the skip files?

Comment (RedBrogdon) [reply web/app.yaml]: It's not listed as an option in the config guide for 3.7:

https://cloud.google.com/appengine/docs/standard/python3/config/appref

I assumed it had been deprecated in favor of .gloudignore, which we could also consider using.

File: web/main.py (hunk 1 above)
Comment (RedBrogdon) [reply web/main.py]: Thanks for the tip!

File: web/app.yaml (hunk 2 above)
Comment (domesticmouse) [reply web/app.yaml]: sgtm

**Pull request merged**